specifications/api/swagger.yaml

```diff
@@ -2940,3 +2997,29 @@ paths:
       parameters:
         - *parametersAddressId
       responses: *responsesInspectAddress
+
+  /settings:
+    put:
+      operationId: putSettings
+      tags: ["Settings"]
+      summary: Settings
+      description: |
+        <p align="right">status: <strong>stable</strong></p>
+
+        Overwrite current settings
```

Comment:
I think more detailed description would be handy explaining what are the possible values and behavior after updating the settings (like stated in https://github.com/input-output-hk/cardano-wallet/issues/2163 in "Decision" section, if it is still up to date.)

Reply:
Since settings should be as general as possible, I don't know how to describe the behavior. That depends on the setting. We can only reasonably document it for the `pool_metadata_source`. That is already done in the appropriate places:

```yaml
x-poolMetadataSource: &poolMetadataSource
  description: |
    Pool metadata source. This sets the metadata fetching strategy.

    Possible values are
      * null -> no fetching
      * direct -> direct fetching
      * uri -> use SMASH server
  oneOf:
    - type: string
      enum:
        - none
        - direct
    - title: URI
      type: string
      format: uri
      example: https://smash.cardano-mainnet.iohk.io/

x-settings: &settings
  description: Settings
  type: object
  required:
    - pool_metadata_source
  properties:
    pool_metadata_source:
      <<: *poolMetadataSource
      description: How to fetch pool metadata
```

Reply:
Ok, perhaps updating to something like:

```
  properties:
    pool_metadata_source:
      <<: *poolMetadataSource
      description: |
        Select metadata fetching strategy:
          - `none` - metadata is not fetched at all,
          - `direct` - metadata is fetched directly from chain,
          - `uri` - metadata is fetched from the external Stake-Pool Metadata Aggregation Server (SMASH)

        After update existing metadata will be dropped forcing it to re-sync automatically with the new setting.
```